components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -1272,57 +1271,69 @@ public Resource getResource(String key) {
      *
+     * @param searchKey String
      * @param folderPath Path of the registry
      * @return JSON object containing the folder structure
      */
-    public JSONObject getRegistryResourceJSON(String folderPath) {
+    public JSONObject getRegistryResourceJSON(String searchKey, String folderPath) {
 
         File node = new File(folderPath);
         JSONObject jsonObject = new JSONObject();
-        addNodesToJSON(node, jsonObject);
-        return jsonObject;
+        addNodesToJSON(searchKey, node, jsonObject);
+
+        JSONObject outputObject = new JSONObject();
+        outputObject.put("list", jsonObject);
+        return outputObject;
     }
 
     /**
-     * Updates the JSON object with existing files and folder in the <MI-HOME>/registry directory.
-     * Updates a MAP with the metadata of each registry.
-     * Updates a MAP with the properties of each registry.
+     * Updates the JSON object with existing files and folder which
+     * match with the searchKey, in the given registry directory.
      *
+     * @param searchKey String
      * @param node       File
      * @param jsonObject JSON object with results
      */
-    private void addNodesToJSON(File node, JSONObject jsonObject) {
+    private void addNodesToJSON(String searchKey, File node, JSONObject jsonObject) {
 
         String nodeName = node.getName();
-        jsonObject.put(NAME_KEY, nodeName);
 
         if (node.isDirectory()) {
-            jsonObject.put(TYPE_KEY, FILE_TYPE_DIRECTORY);
             JSONArray childArray = new JSONArray();
             String[] childNodes = node.list();
             if (childNodes != null) {
                 for (String childNode : childNodes) {
                     if (isNodeNotRequiredToBeFetched(childNode)) {
                         JSONObject nodeJSONObject = new JSONObject();
                         File childFile = new File(node, childNode);
-                        addNodesToJSON(childFile, nodeJSONObject);
-                        childArray.put(nodeJSONObject);
+                        addNodesToJSON(searchKey, childFile, nodeJSONObject);
+                        if (nodeJSONObject.has(NAME_KEY)) {
+                            childArray.put(nodeJSONObject);
+                        }
                     } else if (childNode.endsWith(PROPERTY_EXTENTION)) {
                         String propertyOwner = childNode.replace(PROPERTY_EXTENTION, "");
-                        if (!Arrays.asList(childNodes).contains(propertyOwner)) {
-                            JSONObject nodeJSONObject = new JSONObject();
-                            nodeJSONObject.put(NAME_KEY, propertyOwner);
-                            nodeJSONObject.put(TYPE_KEY, PROPERTY_FILE_VALUE);
-                            nodeJSONObject.put(CHILD_FILES_LIST_KEY, Collections.<String>emptyList());
-                            childArray.put(nodeJSONObject);
+                        if (propertyOwner.toLowerCase().contains(searchKey)) {
```

Comment:
Can propertyOwner become null?

Reply:
it may not. Since childnodes are the list of files that resides in a non null directory